### PR TITLE
Adjust grenade values

### PR DESCRIPTION
**Pull request by Brian-Otten**
#### Summary

 ```SUMMARY: Balance "Adjust grenade values "```

#### Purpose of change
Normal military frag grenades now use the same format as other explosives so the code is less confusing. Adjusted values to match the m67 frag grenade.

#### Describe the solution
https://en.wikipedia.org/wiki/M67_grenade
Explosives power is TNT equivalence, m67 frag has 180 grams of compound B which is 240 grams tnt.
Casing mass used is weight of the grenade minus weight of the explosive filler.

Fragment size was estimated with some help from dev-general at 0.15 grams. This is subject to change if someone has better data / a better way of estimating shrapnel size.

#### Testing
Spawn grenades ingame, they function as expected of military grenades without errors.

#### Additional context
Estimations by tenmillimaster
"my hairy inverted golfball has 726 points, by my count
image

tiny bit of error on the left hand side, but eh
good enough
so if it fragmented perfectly, less fuze elements and whatnot, it'd split into ~1452 pieces?"

The grenade is currently set to fragment in 1466 pieces.

### CODE REVIEW

**Comment by Brian-Otten** [comment]
It looks like adjusting the grenade to match its real life values is making it fail the explosion test, going to try seeing if it passes if we assume it fragmenting less evenly and produce less/larger fragments. If that doesn't pass this PR can probably be closed/stalled until someone reworks the explosion tests.

**Comment by Brian-Otten** [comment]
It looks like the test is failing because it thinks the new grenade is too weak, while it now behaves much more like the other ingame explosives. I "suspect" that with the old code it is not generating 212 pieces of shrapnel, but setting the casing mass to 212 and generating a "lot" of tiny shrapnel
image
Which is extremely effective against completely unarmored zombies but extremely weak against even small armor values.

**Comment by Brian-Otten** [comment]
Kevin helped me figure out that grenade is using the default shrapnel size of 0.005 grams, which gives the frag grenade 42400 pieces of shrapnel, which is probably too much. 
Adjusting the shrapnel size to somewhere closer to 1500 is easily done, but we need to adjust the explosion_balance_test.cpp to accept the new values. 
First idea was to change it to count wounded zombies as "casualties" in the lethality test.

**Comment by Brian-Otten** [comment]
New value seems to give very sensible results, passing the 5m test, and i believe would pass the 15m test if we count the wounded as casualties.

It does seem to be failing the vehicle test now though, i think because the shrapnel now does enough damage to harm the vehicle, which it could not before.

**Comment by Brian-Otten** [comment]
Added the modified explosions test kevin wrote to this PR.

**Comment by Brian-Otten** [comment]
Going to tweak fragment damage to vehicles till i've walked it to the target of narrowly passing the vehicletest again.

**Comment by Brian-Otten** [comment]
Explosion test is now passing lethality/casualty tests, but failing the vehicle test. The frag grenade is slightly harming the wheels and the alternator. It only passes if it does not damage the vehicle. Not sure what the best approach to fix it would be.

**Comment by Brian-Otten** [comment]
... i fat fingered the close PR button. Working on a fix for the vehicle test with suggestions from kevin, once that is narrowed down to pass the test this should be ready.

**Comment by Brian-Otten** [comment]
I'm starting to suspect that the code for reducing damage to vehicles is working oddly.
I've been trying to dial it in so the new grenade does not harm vehicles at all, but with the new fragment sizes it seems to damage vehicles, even when i've changed it to
                g->m.bash( target, damage / 4000, true );
from                 g->m.bash( target, damage / 10, true );

**Comment by Brian-Otten** [comment]
All right, it took some tuning but i finally got this where i want it / where it should be. Grenades now have realistic values and effects, and shrapnel damage to vehicles/objects is behaving as i might expect, with the grenade shrapnel barely damaging vehicles / doors (unless detonated from really close) but heavier explosives doing a bit better.
Little showcase:
Before detonation 
image
After detonation
image

I think this should be ready to merge now. We might want to consider waiting for after 0.E to be fully sure there are no weird interactions in the upcoming stable, but making the grenade handle properly might also be seen as important, so i leave that up to the devs.

File: tests/explosion_balance_test.cpp [hunk 1]
```diff
@@ -125,11 +134,11 @@ static void check_vehicle_damage( const std::string &explosive_id, const std::st
...
-    check_lethality( "grenade_act", 15, 0.40, 0.06 );
+    check_lethality( "grenade_act", 5, 0.95, 0.06, outcome_type::Kill );
+    check_lethality( "grenade_act", 15, 0.40, 0.06, outcome_type::Casualty );
 }
 
 TEST_CASE( "grenade_vs_vehicle", "[grenade],[explosion],[balance]" )
 {
     check_vehicle_damage( "grenade_act", "car", 5 );
-}
+}
```
Comment (kevingranade) [inline tests/explosion_balance_test.cpp]:
```suggestion
}

```

**Comment by Brian-Otten** [comment]
Thanks for the review and teamwork on this one kevin.

**Pull request merged**